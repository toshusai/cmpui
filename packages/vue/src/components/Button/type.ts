export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
