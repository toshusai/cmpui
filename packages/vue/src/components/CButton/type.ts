export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
