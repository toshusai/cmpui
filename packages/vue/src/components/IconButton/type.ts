export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
