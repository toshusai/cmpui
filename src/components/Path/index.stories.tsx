import type { Meta, StoryObj } from "@storybook/react";
import { Path } from ".";

const meta: Meta<typeof Path> = {
  title: "Controls/Path",
  component: Path,
};

export default meta;
type Story = StoryObj<typeof Path>;

export const Basic: Story = {
  render: function Render() {
    return (
      <Path
        commands={[
          { type: "move", x: 0, y: 0 },
          { type: "cubic", x: 200, y: 200, x1: 100, y1: 0, x2: 100, y2: 200 },
        ]}
      />
    );
  },
};
