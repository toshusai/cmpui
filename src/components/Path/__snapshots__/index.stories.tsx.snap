// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/Path Basic smoke-test 1`] = `
<div style="width: 200px; height: 200px; position: relative;">
  <svg stroke="var(--cmpui-primary-color)"
       width="0"
       height="0"
       viewbox="0 0 0 0"
       class="cmpui_curve-controls-svg"
       style="position: absolute; overflow: visible; left: 0px; top: 0px; width: 0px; height: 0px;"
  >
    <path d="M 0 0"
          stroke-width="1"
          fill="none"
    >
    </path>
    <path d="M 0 0 C 100 0 100 200 200 200"
          stroke-width="1"
          fill="none"
    >
    </path>
  </svg>
</div>
`;
